src/cryptonote_core/service_node_deregister.cpp

```diff
@@ -731,4 +795,32 @@ namespace loki
 
     return true;
   }
+
+  void deregister_vote_pool::remove_expired_votes(uint64_t height)
+  {
+    uint64_t const ALIVE_HEIGHT_WINDOW = 20;
+    if (height < ALIVE_HEIGHT_WINDOW)
+    {
+      return;
+    }
+
+    CRITICAL_REGION_LOCAL(m_lock);
+    const time_t now                  = time(NULL);
+    const time_t ALIVE_SECONDS_WINDOW = DIFFICULTY_TARGET_V2 * ALIVE_HEIGHT_WINDOW;
+
+    uint64_t minimum_height = height - ALIVE_HEIGHT_WINDOW;
+    for (auto it = m_deregisters.begin(); it != m_deregisters.end();)
+    {
+      time_t lifetime = now - it->time_group_created;
+      if (it->block_height < minimum_height && lifetime >= ALIVE_SECONDS_WINDOW)
```

Comment:
Thoughts? I have votes living in the pool for 20 blocks but also keep them around if they haven't been alive for at least (20 blocks * target solve time) seconds. 

So if hash rate drops crazy and blocks mined slows down, we don't throw away votes. If blocks get mined quick but the votes haven't lived long enough they stick around so we can reuse the data on reorg (i.e. timing attacks)